File: apps/api/src/main/java/com/lifttrack/api/domain/models/WorkoutSessionExercise.java
package com.lifttrack.api.domain.models;

import java.time.Instant;
import java.util.UUID;

public record WorkoutSessionExercise(
        UUID uuid,
        UUID workoutSessionUuid,
        UUID exerciseUuid,
        int sortOrder,
        String notes,
        Instant createdAt
) {
}

File: apps/web/src/domain/models/workout-session-exercise.ts
export interface WorkoutSessionExercise {
  readonly uuid: string
  readonly workoutSessionUuid: string
  readonly exerciseUuid: string
  readonly sortOrder: number
  readonly notes: string | null
  readonly createdAt: string
}

export interface CreateWorkoutSessionExercise {
  readonly exerciseUuid: string
  readonly sortOrder: number
  readonly notes?: string
}
